c2corg_ui/static/js/cotometer.js

```diff
@@ -0,0 +1,179 @@
+goog.provide('app.CotometerDirective');
+goog.provide('app.CotometerController');
+
+goog.require('app');
+
+/**
+ * This directive is used to manage the dialog of cotometer
+ * cotometerRating function was developed and made available with the support of BLMS
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.CotometerDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppCotometerController',
+    controllerAs: 'cotmetCtrl',
+    templateUrl: '/static/partials/cotometer.html',
+    bindToController: {
+      'module': '<',
+      'lang': '@'
+    }
+  };
+};
+
+app.module.directive('appCotometer', app.CotometerDirective);
+
+
+/**
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {ui.bootstrap.$modalStack} $uibModalStack $uibModalStack.
+ * @constructor
+ * @ngInject
+ */
+app.CotometerController = function(gettextCatalog, $uibModalStack) {
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.rating;
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.skiability;
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.slope;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.errorSlope;
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.elevation;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.errorElevation;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {ui.bootstrap.$modalStack}
+   * @private
+   */
+  this.$uibModalStack_ = $uibModalStack;
+
+  /**
+   * @export
+   * @type {string}
+   */
+  this.module;
+
+  /**
+   * @export
+   * @type {string}
+   */
+  this.lang;
+};
+
+
+/**
+ * @private
+ */
+app.CotometerController.prototype.cotometerRating = function() {
```

Comment:
If the function is private, naming conventions imply a trailing underscore:
``app.CotometerController.prototype.cotometerRating_``

For instance https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/static/js/gpxupload.js#L64